pkg/storage/elasticsearch_secrets.go

```diff
@@ -145,3 +190,25 @@ func getFileContents(path string) []byte {
 	}
 	return contents
 }
+
+func writeToWorkingDirFile(dir, toFile string, value []byte) error {
+	// first check if file exists - we prefer what is on FS to revert users editing secrets
+	path := getFilePath(dir, toFile)
+	if _, err := os.Stat(path); err == nil {
+		return nil
+	}
+	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
```

Comment:
Thought this method would only be called for overwriting the secret data with existing secret - so shouldn't the file already exist? Not a big issue, but just wanted to clarify my understanding.

Reply:
This method is called on every loop. The previous statement returns if file exists. If the file does not exists we have to create before creating the file.